### ISSUE TITLE
can't find -lblas or -lcblas

### ISSUE THREAD

**Issue by nbecker** [serge-sans-paille/pythran#592]
current git version isn't working on fedora F24.  I have openblas, and could possibly link with mkl blas, but no cblas

```
pythran3 mag_sqr.cpp
WARNING: Disabling color, you really want to install colorlog.
/usr/bin/ld: cannot find -lcblas
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -lcblas
collect2: error: ld returned 1 exit status
/tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o: In function `main':
source.c:(.text+0x96): undefined reference to `cblas_ddot'
collect2: error: ld returned 1 exit status
/tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o: In function `main':
source.c:(.text+0x96): undefined reference to `cblas_ddot'
collect2: error: ld returned 1 exit status
Traceback (most recent call last):
  File "/usr/lib64/python3.5/distutils/unixccompiler.py", line 204, in link
    self.spawn(linker + ld_args)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/ccompiler.py", line 27, in <lambda>
    m = lambda self, *args, **kw: func(self, *args, **kw)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/ccompiler.py", line 71, in CCompiler_spawn
    raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
distutils.errors.DistutilsExecError: Command "cc /tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o -L/usr/lib64 -lcblas -o /tmp/tmppprhr2hh/a.out" failed with exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/system_info.py", line 1716, in has_cblas
    extra_postargs=info.get('extra_link_args', []))
  File "/usr/lib64/python3.5/distutils/ccompiler.py", line 734, in link_executable
    debug, extra_preargs, extra_postargs, None, target_lang)
  File "/usr/lib64/python3.5/distutils/unixccompiler.py", line 206, in link
    raise LinkError(msg)
distutils.errors.LinkError: Command "cc /tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o -L/usr/lib64 -lcblas -o /tmp/tmppprhr2hh/a.out" failed with exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.5/distutils/unixccompiler.py", line 204, in link
    self.spawn(linker + ld_args)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/ccompiler.py", line 27, in <lambda>
    m = lambda self, *args, **kw: func(self, *args, **kw)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/ccompiler.py", line 71, in CCompiler_spawn
    raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
distutils.errors.DistutilsExecError: Command "cc /tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o -L/usr/lib64 -lblas -o /tmp/tmppprhr2hh/a.out" failed with exit status 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nbecker/.local/bin/pythran3", line 9, in <module>
    load_entry_point('pythran==0.7.6.post1', 'console_scripts', 'pythran')()
  File "/home/nbecker/.local/lib/python3.5/site-packages/pythran-0.7.6.post1-py3.5.egg/pythran/run.py", line 126, in run
    **compile_flags(args))
  File "/home/nbecker/.local/lib/python3.5/site-packages/pythran-0.7.6.post1-py3.5.egg/pythran/toolchain.py", line 236, in compile_cxxfile
    extension_args = make_extension(**kwargs)
  File "/home/nbecker/.local/lib/python3.5/site-packages/pythran-0.7.6.post1-py3.5.egg/pythran/config.py", line 81, in make_extension
    numpy_blas = numpy_sys.get_info("blas")
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/system_info.py", line 372, in get_info
    return cl().get_info(notfound_action)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/system_info.py", line 560, in get_info
    self.calc_info()
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/system_info.py", line 1679, in calc_info
    lib = self.has_cblas(info)
  File "/usr/lib64/python3.5/site-packages/numpy/distutils/system_info.py", line 1722, in has_cblas
    extra_postargs=info.get('extra_link_args', []))
  File "/usr/lib64/python3.5/distutils/ccompiler.py", line 734, in link_executable
    debug, extra_preargs, extra_postargs, None, target_lang)
  File "/usr/lib64/python3.5/distutils/unixccompiler.py", line 206, in link
    raise LinkError(msg)
distutils.errors.LinkError: Command "cc /tmp/tmppprhr2hh/tmp/tmppprhr2hh/source.o -L/usr/lib64 -lblas -o /tmp/tmppprhr2hh/a.out" failed with exit status 1
```

**Comment by nbecker** [comment]
I see this is an issue with python3.5.  I was using
pythran -E mag_sqr.py
pythran3 mag_sqr.cpp (where pythran3 is the pythran script modified as
#!/usr/bin/python3)

**Comment by nbecker** [comment]
problem is coming from config.py:
It's not just F24 numpy though, I tried anaconda version and get same result on py35:

import numpy.distutils.system_info as ns

In [16]: ns
Out[16]: 
<module 'numpy.distutils.system_info' from '/home/nbecker/anaconda3/envs/py35/lib/python3.5/site-packages/numpy/distutils/system_info.py'>

In [17]: ns.get_info('blas')
/usr/bin/ld: cannot find -lcblas
collect2: error: ld returned 1 exit status
/usr/bin/ld: cannot find -lcblas

**Comment by serge-sans-paille** [comment]
On Wed, Jul 20, 2016 at 05:58:10AM -0700, ndbecker wrote:
> I see this is an issue with python3.5.  I was using
> pythran -E mag_sqr.py
> pythran3 mag_sqr.cpp (where pythran3 is the pythran script modified as
> #!/usr/bin/python3)

Hi Neal and thanks for the report. I think it's a perfect opportunity to
try our beta native Python3 support, from branch feature/python-3. Can
you have a look ;-)

**Comment by nbecker** [comment]
Thanks!

Same result (-lcblas).

For now, I disable blas check in config.py and hardcode:
    # blas dependency
    # numpy_blas = numpy_sys.get_info("blas")
    # extension['libraries'].extend(numpy_blas.get('libraries', []))
    # extension['library_dirs'].extend(numpy_blas.get('library_dirs', []))
    extension['libraries'].extend(['openblas'])

**Comment by serge-sans-paille** [comment]
On Wed, Jul 20, 2016 at 06:30:57AM -0700, ndbecker wrote:
> Thanks!
> 
> Same result (-lcblas).
> 
> For now, I disable blas check in config.py and hardcode:
>     # blas dependency
>     # numpy_blas = numpy_sys.get_info("blas")
>     # extension['libraries'].extend(numpy_blas.get('libraries', []))
>     # extension['library_dirs'].extend(numpy_blas.get('library_dirs', []))
>     extension['libraries'].extend(['openblas'])

Then the best thing to do may be to add a blas entry in the config file.
It would override the numpy default if given...

**Comment by serge-sans-paille** [comment]
@nbecker #593 makes it possible to customize the entry from the config file.

**Comment by serge-sans-paille** [comment]
Fixed by  #593
